AstarteDeviceSDKCSharp/Device/AstarteDevice.cs

```diff
@@ -42,6 +42,30 @@ public class AstarteDevice : IAstarteTransportEventListener
         private static int MIN_INCREMENT_INTERVAL = 5000;
         private static int MAX_INCREMENT_INTERVAL = 60000;
 
+        /// <summary>
+        /// Basic class to define an Astarte Device.
+        /// Device represents an Astarte Device. It is the base class used for managing the Device
+        /// lifecycle and data. Users should instantiate a Device with the right credentials 
+        /// and connect it to the configured instance to start working with it. 
+        /// </summary>
+        /// <param name="deviceId">The Device ID for this Device. 
+        /// It has to be a valid Astarte Device ID.
+        /// </param>
+        /// <param name="astarteRealm">The Realm this Device will be connecting against.</param>
+        /// <param name="credentialSecret">The Credentials Secret for this Device. The Device 
+        ///    class assumes your Device has
+        ///    already been registered - if that is not the case, you can use either
+        ///    :py:func:`register_device_with_jwt_token` or
+        ///    :py:func:`register_device_with_private_key`.
+        /// </param>
+        /// <param name="astarteInterfaceProvider">Class for loading Astarte interfaces</param>
+        /// <param name="pairingBaseUrl">The Base URL of Pairing API of the 
+        /// Astarte Instance the Device will connect to.</param>
+        /// <param name="cryptoStoreDirectory">Path to an existing directory which will be used
+        /// for holding persistency for this
+        /// device: certificates, caching and more. It doesn't have to be unique per device,
+        /// a subdirectory for the given Device ID will be created.
+        /// </param>
```

Comment:
```suggestion
        /// <summary>
        /// Basic class defining an Astarte device.
        /// Used for managing the device lifecycle and data. 
        /// Users can instantiate a device by providing a set of credentials 
        /// and connect it to the Astarte instance. 
        /// </summary>
        /// <param name="deviceId">The device ID for this device. 
        /// It has to be a valid Astarte device ID.
        /// </param>
        /// <param name="astarteRealm">The realm this device will be connecting to.</param>
        /// <param name="credentialSecret">The credentials secret for this device. This class assumes your device has
        ///    already been registered. If that is not the case register your device using either
        ///    :py:func:`register_device_with_jwt_token` or
        ///    :py:func:`register_device_with_private_key`.
        /// </param>
        /// <param name="astarteInterfaceProvider">Class for loading Astarte interfaces</param>
        /// <param name="pairingBaseUrl">The base URL of the pairing API of the 
        /// Astarte instance the device will connect to.</param>
        /// <param name="cryptoStoreDirectory">Path to an existing directory which will be used
        /// to store the persistent data of this device. (i.e. certificates, caching, and more). 
        /// It can be a shared directory for multiple devices, a subdirectory for the given device ID will be created.
        /// </param>
```